### PR TITLE
turns hover over tooltips for inventory items into a preference

### PR DESCRIPTION
**Pull request by roarbark**
## What Does This PR Do

this allows people to toggle the tooltip boxes that appear when hovering over items in your inventory

## Why It's Good For The Game

the tooltip boxes are not useful at all for people like me, who naturally use shift click to read item descriptions.
however, since they are completely opaque and do not scale with UI alpha preference they tend to block a bit of the game window- this is especially noticeable when hovering your mouse over a laser gun in your suit storage ready to draw it while patrolling maintenance. since the laser gun doesn't have a short examine text the box can occlude up to 5 tiles.

it's annoying and redundant- for players in more mechanical roles

for players that care less about edge of screen sight or would rather use the tooltips this is implemented as a preference that can be toggled on or off.

## Images of changes

https://user-images.githubusercontent.com/41054538/180857346-735c5c7c-6322-4617-8fba-3941cf138a6d.mp4

## Changelog
:cl:
add: preference option to toggle item description boxes that appear after hovering over an inventory item
/:cl:

### CODE REVIEW

File: code/__DEFINES/preferences.dm [hunk 1]
```diff
@@ -57,6 +57,7 @@
 #define PREFTOGGLE_2_FORCE_WHITE_RUNECHAT	1024
 #define PREFTOGGLE_2_SIMPLE_STAT_PANEL	2048
 #define PREFTOGGLE_2_SEE_ITEM_OUTLINES 	4096
+#define PREFTOGGLE_2_SEE_ITEM_TOOLTIPS  8192
 
 #define TOGGLES_2_TOTAL 			8191 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
```
Comment (Sirryan2002) [inline code/__DEFINES/preferences.dm:62]: please make sure to update this

Comment (roarbark) [reply code/__DEFINES/preferences.dm]: guh!

File: code/__DEFINES/preferences.dm [hunk 2]
```diff
@@ -57,8 +57,9 @@
 #define PREFTOGGLE_2_FORCE_WHITE_RUNECHAT	1024
 #define PREFTOGGLE_2_SIMPLE_STAT_PANEL	2048
 #define PREFTOGGLE_2_SEE_ITEM_OUTLINES 	4096
+#define PREFTOGGLE_2_SEE_ITEM_TOOLTIPS  8192
 
-#define TOGGLES_2_TOTAL 			8191 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
+#define TOGGLES_2_TOTAL 			12288 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
```
Comment (S34NW) [inline code/__DEFINES/preferences.dm:62]:
```suggestion
#define TOGGLES_2_TOTAL 			16383 // If you add or remove a preference toggle above, make sure you update this define with the total value of the toggles combined.
```

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -715,12 +715,13 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /obj/item/MouseEntered(location, control, params)
 	. = ..()
 	if(in_inventory || in_storage)
-		var/timedelay = 8
 		var/mob/user = usr
-		tip_timer = addtimer(CALLBACK(src, .proc/openTip, location, control, params, user), timedelay, TIMER_STOPPABLE)
+		var/mob/living/L = user
```
Comment (S34NW) [inline code/game/objects/items.dm:719]: Why do you move this line up here? It would be better to put it directly above line 727

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/__DEFINES/preferences.dm (hunk 2 above)
Comment (S34NW) [reply code/__DEFINES/preferences.dm:62]: (final toggle * 2) - 1

**Comment by Contrabang** [comment]
Does this affect tooltips for radial menus? IIRC they look very similar and might share some of the code

File: code/__DEFINES/preferences.dm [hunk 4]
```diff
@@ -57,8 +57,9 @@
 #define PREFTOGGLE_2_FORCE_WHITE_RUNECHAT	1024
 #define PREFTOGGLE_2_SIMPLE_STAT_PANEL	2048
 #define PREFTOGGLE_2_SEE_ITEM_OUTLINES 	4096
+#define PREFTOGGLE_2_SEE_ITEM_TOOLTIPS  8192
```
Comment (AffectedArc07) [inline code/__DEFINES/preferences.dm:60]:
```suggestion
#define PREFTOGGLE_2_HIDE_ITEM_TOOLTIPS  8192
```
Enabling this turns it off, rename accordingly.

File: code/__DEFINES/preferences.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/__DEFINES/preferences.dm:62]: This needs doing

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 3 above)
Comment (roarbark) [reply code/game/objects/items.dm:719]: uhh, stupid habit from another language i was taught in school- i'll change it

File: code/__DEFINES/preferences.dm (hunk 2 above)
Comment (roarbark) [reply code/__DEFINES/preferences.dm]: oops

**Comment by roarbark** [comment]
> Does this affect tooltips for radial menus? IIRC they look very similar and might share some of the code

good question, it doesn't. though i feel its better that way, since its more contextual, has much smaller boxes and is much more responsive, since you dont need to wait for the boxes to appear.

File: code/__DEFINES/preferences.dm (hunk 4 above)
Comment (roarbark) [reply code/__DEFINES/preferences.dm]: yeah, makes more sense

**Review by SteelSlayer — APPROVED** [review]

File: code/modules/client/preference/preferences_toggles.dm [hunk 5]
```diff
@@ -342,6 +342,14 @@
 	prefs.save_preferences(src)
 	to_chat(usr, "You will [(prefs.toggles2 & PREFTOGGLE_2_SEE_ITEM_OUTLINES) ? "now" : "no longer"] see item outlines on hover.")
 
+/client/verb/toggle_item_tooltips()
+	set name = "Toggle Hover-over Item Tooltips"
+	set category = "Preferences"
+	set desc = "Toggles textboxes with the item descriptions after hovering on them in your inventory."
+	prefs.toggles2 ^= PREFTOGGLE_2_HIDE_ITEM_TOOLTIPS
+	prefs.save_preferences(src)
+	to_chat(src, "You will [(prefs.toggles2 & PREFTOGGLE_2_HIDE_ITEM_TOOLTIPS) ? "no longer" : "now"] see item tooltips on hover.")
```
Comment (AffectedArc07) [inline code/modules/client/preference/preferences_toggles.dm:351]:
```suggestion
	to_chat(src, "You will [(prefs.toggles2 & PREFTOGGLE_2_HIDE_ITEM_TOOLTIPS) ? "no longer" : "now"] see item tooltips when you hover items on your HUD.")
```
Better distinction.

File: code/game/objects/items.dm [hunk 6]
```diff
@@ -715,14 +715,15 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /obj/item/MouseEntered(location, control, params)
 	. = ..()
 	if(in_inventory || in_storage)
-		var/timedelay = 8
 		var/mob/user = usr
-		tip_timer = addtimer(CALLBACK(src, .proc/openTip, location, control, params, user), timedelay, TIMER_STOPPABLE)
+		if(!(user.client.prefs.toggles2 & PREFTOGGLE_2_HIDE_ITEM_TOOLTIPS))
+			var/timedelay = 8
+			tip_timer = addtimer(CALLBACK(src, .proc/openTip, location, control, params, user), timedelay, TIMER_STOPPABLE)
```
Comment (AffectedArc07) [inline code/game/objects/items.dm:721]:
```suggestion
			tip_timer = addtimer(CALLBACK(src, .proc/openTip, location, control, params, user), 8, TIMER_STOPPABLE)
```

File: code/modules/client/preference/preferences_toggles.dm (hunk 5 above)
Comment (roarbark) [reply code/modules/client/preference/preferences_toggles.dm]: agreed, i made it say "hover over" though

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**